File: Selection_Sort.py
# selection sort 

def selection_sort(L):
    n = len(L)
    for i in range(n - 1):
        min_pos = i
        for j in range(i + 1, n):
            if L[j] < L[min_pos]:
                min_pos = j
        temp = L[i]
        L[i] = L[min_pos]
        L[min_pos] = temp
    return L

File: Selection_Sort.c
// Selection Sort: best case: O(n^2)
//                 worst case: O(n^2)
// mutate array

void selection_sort(int arr[], int len) {
    for (int i = 0; i < len - 1; i++) {  // only need to loop from the very first element to the second last element
        int min_pos = i;
        for (int j = i + 1; j < len; j++) {
            if (arr[j] < arr[min_pos] {
                min_pos = j;
            }
        }
        int temp = arr[i];
        arr[i] = arr[min_pos];
        arr[min_pos] = temp;
    }
}
